### ISSUE TITLE
scanf display regression in crackme0x00.exe

### ISSUE THREAD

**Issue by Maijin** [wargio/r2dec-js#173]
Actually not sure if regression or if it always happened but notice the scanf when doing aaa / pdd in crackme0x00.exe (r2r/bins/pe/ioli)

```
scanf (0x404024);
```

Instead should be:

```
scanf ("%s", &s1);
```

**Comment by wargio** [comment]
Hmm maybe a regression. I'll check later

**Comment by wargio** [comment]
fixed scanf with commit https://github.com/wargio/r2dec-js/commit/9b27dfdb33cc75c9fc5cef3c4c3c4d7593ad5033

```
[0x00401310]> Cs 3 @ 0x404024
[0x00401310]> pdd
/* r2dec pseudo code output */
/* ~/radare2-regressions/bins/pe/crackme0x00.exe @ 0x401310 */
#include <stdint.h>
 
int32_t main (void) {
    int32_t var_1ch;
    char * s1;
    char * s2;
    eax = 0;
    eax += 0xf;
    eax += 0xf;
    eax >>= 4;
    eax <<= 4;
    var_1ch = eax;
    fcn_00402c70 ();
    _main ();
    printf ("IOLI Crackme Level 0x00\n");
    printf ("Password: ");
    eax = &s1;
    scanf ("%s", eax);
    eax = &s1;
    eax = _strcmp (eax, "250382");
    if (eax != 0) {
        printf ("Invalid Password!\n");
    } else {
        printf ("Password OK :)\n");
    }
    eax = 0;
    return eax;
}

```

**Comment by Maijin** [comment]
Hum, I still have `scanf (0x404024, eax);`

**Comment by radare** [comment]
Did you do af or aaa?

**Comment by Maijin** [comment]
same result both.

**Comment by wargio** [comment]
@Maijin `Cs 3 @ 0x404024`
the string is too short to be detected with default options

**Comment by Maijin** [comment]
those %XX should be handled by either r2dec or radare2, I think in radare2 it will be hard because of this default option, but r2dec could automatically have some routines to detect %XX strings no?

**Comment by wargio** [comment]
yes i could, but i would like to let the user to choose what to do.

**Comment by Maijin** [comment]
Hum how would that be useful for a user to not show scanf/printf format strings 🤔

**Comment by wargio** [comment]
hmmm... i could add that logic, but not so sure how to handle it. for example how about native android logger? should i support that too? it might be complex..

**Comment by Maijin** [comment]
Can always start small and enhance later, worth giving it a shot imho because those are very prevalent.

**Comment by wargio** [comment]
okok.

**Comment by elicn** [comment]
I agree with @wargio, this is a r2 configuration that should be tweaked per user preference.

**Comment by Maijin** [comment]
But changing this introduce tons of false positives on radare2 and inherently r2dec side though.

**Comment by wargio** [comment]
what i mean is also on other decompilers you have the user to choose to try to convert an address to string, etc..

**Comment by Maijin** [comment]
Sure but those scanf/printf cases work out of the boxes in all other decompilers :p
